x-pack/plugins/maps/public/layers/sources/source.d.ts

```diff
@@ -18,13 +29,16 @@ export interface ISource {
   isQueryAware(): boolean;
   isRefreshTimerAware(): Promise<boolean>;
   isTimeAware(): Promise<boolean>;
+  getImmutableProperties(): Promise<ImmutableSourceProperty[]>;
+  getAttributions(): Promise<Attribution[]>;
 }
 
 export class AbstractSource implements ISource {
-  constructor(sourceDescriptor: AbstractSourceDescriptor, inspectorAdapters: object);
+  readonly _descriptor: AbstractSourceDescriptor;
+  constructor(sourceDescriptor: AbstractSourceDescriptor, inspectorAdapters?: object);
 
   destroy(): void;
-  createDefaultLayer(): ILayer;
+  createDefaultLayer(options?: LayerDescriptor, mapColors?: string[]): ILayer;
```

Comment:
Should this also be Partial<LayerDescriptor> here?

Reply:
Good point. Fwiw, I'd leave it for now. In the code, it's either omitted, or preceded by a `Layer.createDescriptor` call, which does the normalization.